runners/core-java/src/test/java/org/apache/beam/runners/core/ReduceFnRunnerTest.java

```diff
@@ -1032,6 +1076,54 @@ public void noEmptyPanesFinalAlways() throws Exception {
         WindowMatchers.isSingleWindowedValue(containsInAnyOrder(1, 2), 9, 0, 10)));
   }
 
+  /**
+   * If the trigger does not care about the watermark, the ReduceFnRunner should still emit an
+   * element for the ON_TIME pane.
```

Comment:
Why? is the default behavior now to always emit an on-time pane? That is problematic -- a user that has a trigger and expects a single pane may now see multiple panes. Consider:

.withTrigger(AfterProcessingTime.pastFirstElementInPane().plusDelayOf(1 day));

I expect exactly one triggering, and I expect it after 1 day has elapsed since I received the first element. With your change, I now need to deal with multiple trigger firings and be pane aware, because I may get an on-time firing before that.